sources/directconnect/virturl-interface.go

```diff
@@ -0,0 +1,198 @@
+			// +overmind:link directconnect-direct-connect-gateway-attachment
+			item.LinkedItemQueries = append(item.LinkedItemQueries, &sdp.LinkedItemQuery{
+				Query: &sdp.Query{
+					Type:   "directconnect-direct-connect-gateway-attachment",
+					Method: sdp.QueryMethod_SEARCH,
+					// returns list of attachments for the given virtual interface id
+					Query: fmt.Sprintf(virtualInterfaceIDFmt, *virtualInterface.VirtualInterfaceId),
+					Scope: scope,
+				},
+				BlastPropagation: &sdp.BlastPropagation{
+					// Changes to the attachment won't affect virtual interface
+					In: false,
+					// If virtual interface is deleted, the attachment state will change to detaching
+					// https://docs.aws.amazon.com/directconnect/latest/APIReference/API_DirectConnectGatewayAttachment.html#API_DirectConnectGatewayAttachment_Contents
+					Out: true,
+				},
+			})
+		}
+
+		items = append(items, &item)
+	}
+
+	return items, nil
+}
+
+//go:generate docgen ../../docs-data
+// +overmind:type directconnect-virtual-interface
+// +overmind:descriptiveType Virtual Interface
+// +overmind:get Get a virtual interface by ID
+// +overmind:list List all virtual interfaces
+// +overmind:search Search virtual interfaces by connection ID
+// +overmind:group AWS
+// +overmind:terraform:queryMap aws_dx_private_virtual_interface.id
+// +overmind:terraform:queryMap aws_dx_public_virtual_interface.id
+// +overmind:terraform:queryMap aws_dx_transit_virtual_interface.id
+
+func NewVirtualInterfaceSource(config aws.Config, accountID string, limit *sources.LimitBucket) *sources.DescribeOnlySource[*directconnect.DescribeVirtualInterfacesInput, *directconnect.DescribeVirtualInterfacesOutput, *directconnect.Client, *directconnect.Options] {
+	return &sources.DescribeOnlySource[*directconnect.DescribeVirtualInterfacesInput, *directconnect.DescribeVirtualInterfacesOutput, *directconnect.Client, *directconnect.Options]{
+		Config:    config,
+		Client:    directconnect.NewFromConfig(config),
+		AccountID: accountID,
+		ItemType:  "directconnect-virtual-interface",
+		DescribeFunc: func(ctx context.Context, client *directconnect.Client, input *directconnect.DescribeVirtualInterfacesInput) (*directconnect.DescribeVirtualInterfacesOutput, error) {
+			limit.Wait(ctx) // Wait for rate limiting
+			return client.DescribeVirtualInterfaces(ctx, input)
+		},
+		InputMapperGet: func(scope, query string) (*directconnect.DescribeVirtualInterfacesInput, error) {
+			return &directconnect.DescribeVirtualInterfacesInput{
+				VirtualInterfaceId: &query,
+			}, nil
+		},
+		InputMapperList: func(scope string) (*directconnect.DescribeVirtualInterfacesInput, error) {
+			return &directconnect.DescribeVirtualInterfacesInput{}, nil
+		},
+		OutputMapper: virtualInterfaceOutputMapper,
+		InputMapperSearch: func(ctx context.Context, client *directconnect.Client, scope, query string) (*directconnect.DescribeVirtualInterfacesInput, error) {
+			if strings.HasPrefix(query, connectionIDPrefix) {
```

Comment:
So this Search implementaion allows you to search by conneciton ID rather than Virtual interface ID, nice. In this case you don't have to do a prefix, the docs comment of `Search virtual interfaces by connection ID` is enough here I think.

Reply:
Thanks for this.

I went by this comment [here](https://github.com/overmindtech/aws-source/pull/364#discussion_r1445746914).

I can change it as recommended as well.

Reply:
Ah I see, let's check: @DavidS-ovm was there a particular reason you suggested a prefix on the search?

Reply:
I was under the impression that there might be more search options than just the connection id, which then needed to be disambiguated.

Reply:
Makes sense @tomasdembelli if there isn't another potential search option, then let's skip the prefix

Reply:
Removed redundant prefixes.
https://github.com/overmindtech/aws-source/pull/373/commits/bd8221b0f3f0f1db7fa096f8e0c4b8e4b8462901